homeassistant/components/discovergy/sensor.py

```diff
@@ -226,10 +162,11 @@ async def async_setup_entry(
 
         if sensors is not None:
             for description in sensors:
-                keys = [description.key] + description.alternative_keys
-
                 # check if this meter has this data, then add this sensor
-                for key in keys:
+                for key in {description.key, *description.alternative_keys}:
+                    coordinator: DiscovergyUpdateCoordinator = data.coordinators[
+                        meter.get_meter_id()
```

Comment:
We can store the meter id in a local variable above as it doesn't change for every entity just for every meter.